### PR TITLE
Feature/iat 3514

### PR DESCRIPTION
**Pull request by bdrainer**
I discussed this with some of you but to recap
- scan item path for media files 
- pull the referenced media files from the item
- if the file on disk is not referenced in item.json then the error is thrown

Notice I scan for files with extension in both lower and upper case.  I wonder if that's necessary or if there is a slicker way to do what I'm doing.  

I wonder if this validation is now different given TIMS and its rules.  A bridge to cross later, but I wonder if there is ever a way for a media file not to be referenced in the item json but does exist on disk.

### CODE REVIEW

File: ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/metadata/MediaTabulator.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+     *
+     * @param item The item to inspect media files for.
+     * @return A list of file names associated with the item.
+     */
+    List<String> collectMediaFilesFromItem(Item item) {
+        List<String> files = new ArrayList<>();
+        files.addAll(getAttachmentFilenamesBySection(item, SECTION_ASL));
+        files.addAll(getAttachmentFilenamesBySection(item, SECTION_AUDIO));
+        files.addAll(getAttachmentFilenamesBySection(item, SECTION_CC));
+        files.addAll(getAttachmentFilenamesBySection(item, SECTION_GLOSSARY));
```
Comment (bdrainer) [inline ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/metadata/MediaTabulator.java]: I should exclude glossary.  I will make that change.

File: ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/metadata/MediaTabulator.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            .map(t -> t.toString().toUpperCase())
+            .collect(Collectors.joining(","));
+
+        String lowerCaseExtensions = Arrays.stream(SaaifConstants.MediaFileTypes.values())
+            .map(t -> t.toString().toLowerCase())
+            .collect(Collectors.joining(","));
+
+        String filenameFilterPattern = String.format("glob:**.{%s,%s}", upperCaseExtensions, lowerCaseExtensions);
+
+        PathMatcher fileFilter = fileSystem.getPathMatcher(filenameFilterPattern);
```
Comment (bdrainer) [inline ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/metadata/MediaTabulator.java]: Here's how I'm searching for both lower and upper case file extensions.

File: ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/SaaifUtil.java [hunk 3]
```diff
@@ -36,4 +36,5 @@ public static String getMachineScoringFileName(final ItemRelease.Item item) {
             ? StringUtils.trimToNull(machineRubricList.get(0).getFilename())
             : StringUtils.EMPTY;
     }
+
```
Comment (malaffoon) [inline ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/SaaifUtil.java]: Dang, i needed that line in my work.

File: ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/metadata/MediaTabulator.java (hunk 2 above)
Comment (malaffoon) [reply ap-item-validation/src/main/java/org/opentestsystem/ap/item/validator/tabulator/metadata/MediaTabulator.java]: So mixed case won't be seen; nbd just commenting.

What folder are we looking through here? Are we expecting gazillions of files? If not, you could just gather all the filenames and then filter in java which is easier.

**Pull request merged**